dotcom-rendering/window.guardian.d.ts

```diff
@@ -20,6 +21,7 @@ declare global {
 			polyfilled: boolean;
 			onPolyfilled: () => void;
 			queue: Array<() => void>;
+			emotionCache?: EmotionCache;
```

Comment:
I'm not sure I love adding to window.guardian but I wasn't sure of another way to keep a single cache between all islands except for relying on a singleton

Reply:
This is your shout obvs but I'd vote to not have shared modules on the window if we can avoid it.

I think a singleton approach has some advantages:
- not reliant on a browser environment
- keeps islands slightly more autonomous and self initialising
- we can track module usage with regular tooling e.g. webpack stats

Would the following also work?:

- create a lib which initialises the emotion cache and exposes a singleton emotionCache
- all islands (and SSR) can use this same lib to get the emotion cache

Reply:
I wonder if this is possible at the moment, since we currently have islands/init.ts and initDiscussion/init.ts as two entrypoints in webpack, we couldn't create a shared library (say emotionCache.ts) and have it shared as a singleton, as it would be included twice by webpack.

Given this fixes an active bug on the site (from my testing), perhaps I should mark this down as a todo for after we complete the single entrypoint work?

Reply:
Ah yes good point we would be at the mercy of webpack's chunking strategy. There's the possibility the module could be duplicated in each chunk. So agree it makes sense to do as a follow up after the single entrypoint update.